lib/addons/src/hooks.ts

```diff
@@ -424,17 +433,21 @@ export function useParameter<S>(parameterKey: string, defaultValue?: S): S | und
 }
 
 /* Returns current value of story args */
-export function useArgs(): [Args, (newArgs: Args) => void, (argNames?: [string]) => void] {
+export function useArgs<TArgs = Args>(): [
```

Comment:
We want `TArgs` to `extends Args` otherwise the user could set it to anything?
```suggestion
export function useArgs<TArgs extends Args = Args>(): [
```